src/Parser/Priest/Discipline/Modules/Items/Estel.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Atonement from '../Spells/Atonement';
+
+const DEBUG = false;
+
+class Estel extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    atonementModule: Atonement,
+  };
+
+  // max number of buffs = 30 (30 player raid)... techincally it's higher
+  // with pets, but getting that many atonments out is nigh impossible.
+  // ideally I think I would use a dictionary... but this works for now while I get it working
+  _timePerHastePercentage = [0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0];
+  lastBuffTimestamp = 0;
+  lastHasteValue = 0;
+
+  avgHaste = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.ESTEL_DEJAHNAS_INSPIRATION.id);
+  }
+
+  get timePerHastePercentage() {
+    return Object.keys(this._timePerHastePercentage).map(key => this._timePerHastePercentage[key]);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId === SPELLS.ESTEL_DEJAHNAS_INSPIRATION_BUFF.id) {
+      console.log(`num atonements found: ${this.atonementModule.numAtonementsActive}`);
+      this.lastHasteValue = this.atonementModule.numAtonementsActive;
```

Comment:
Try to avoid using the atonement module, it does not always give the correct count unless @rp4rk fixed it

Reply:
I will look into this.

Reply:
@karlpralow I think that's a weird recommendation; having the best logic to track active Atonements in one module would seem better to me than having different Atonement tracking mechanisms across multiple modules. Even if the Atonement module is off, why would another implementation be better, and better yet why reinvent the wheel when you could fix the one you already have?